components/EmotionCardContent.js

```diff
@@ -1,10 +1,33 @@
 import styled from "styled-components";
 import { emotionsIcons } from "./EmotionIcons";
 
-export default function EmotionCardContent({ emotion }) {
+export default function EmotionCardContent({ emotion, emotionTypes }) {
+  const matchedEmotionIcon = (() => {
+    // Hole die zugehörige emotionIcon-ID aus emotionTypes
+    const customEmotion = emotionTypes.find(
+      (item) => item.emotionType === emotion.emotionType
+    );
+    if (customEmotion?.emotionIcon) {
+      // Finde das passende Icon in emotionsIcons basierend auf der ID
+      return emotionsIcons.find(
+        (icon) => icon.id === customEmotion.emotionIcon
+      );
+    }
+    // Fallback: Nutze emotionType als Schlüssel, falls keine emotionIcon-ID definiert ist
+    return emotionsIcons.find(
+      (icon) => icon.emotionType === emotion.emotionType
+    );
+  })();
+
+  const matchedEmotionColor = emotionTypes.find(
+    (color) => color.emotionType === emotion.emotionType
+  );
```

Comment:
this was already found in the component EmotionCard. Please pass down the information as a prop.

Reply:
refactored